akka-http-core/src/main/scala/akka/http/impl/engine/ws/Handshake.scala

```diff
@@ -57,69 +64,202 @@ private[http] object Handshake {
             list of values indicating which extensions the client would like
             to speak.  The interpretation of this header field is discussed
             in Section 9.1.
-   */
-  def isWebsocketUpgrade(headers: List[HttpHeader], hostHeaderPresent: Boolean): Option[UpgradeToWebsocket] = {
-    def find[T <: HttpHeader: ClassTag]: Option[T] =
-      headers.collectFirst {
-        case t: T ⇒ t
+    */
+    def isWebsocketUpgrade(headers: List[HttpHeader], hostHeaderPresent: Boolean): Option[UpgradeToWebsocket] = {
+      def find[T <: HttpHeader: ClassTag]: Option[T] =
+        headers.collectFirst {
+          case t: T ⇒ t
+        }
+
+      // val host = find[Host]
+      val upgrade = find[Upgrade]
+      val connection = find[Connection]
+      val key = find[`Sec-WebSocket-Key`]
+      val version = find[`Sec-WebSocket-Version`]
+      // val origin = find[Origin]
+      val protocol = find[`Sec-WebSocket-Protocol`]
+      val supportedProtocols = protocol.toList.flatMap(_.protocols)
+      // FIXME: support extensions
+      // val extensions = find[`Sec-WebSocket-Extensions`]
```

Comment:
Ok, sounds reasonable to me.
I ticketified it explicitly, in case someone has an actual need for it there's a ticket - https://github.com/akka/akka/issues/18709 (I currently don't see the need to jump in on them right away)